src/v/features/feature_table.h

```diff
@@ -77,6 +77,7 @@ enum class feature : std::uint64_t {
     cluster_topic_manifest_format_v2 = 1ULL << 45U,
     node_local_core_assignment = 1ULL << 46U,
     unified_tx_state = 1ULL << 47U,
+    group_fence_batch_type_switch = 1ULL << 48U,
```

Comment:
nit: a bit unclear to me what "type switch" here means? I guess the incompatible change here is that we are using a new batch type, and it would be nice to consistently use a single name in all the places that are referring to it.